### PR TITLE
TE-2639 Support for named release images

### PR DESCRIPTION
**Pull request by jmbowman**
Use [docker-compose variable substitution](https://docs.docker.com/compose/compose-file/#variable-substitution) to more easily support using devstack Docker images for named Open edX releases.

We'll want to hold off on merging this until ``edx-e2e-tests`` has been tagged for release and we have a full set of ``hawthorn.master`` images so we can test it properly.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -351,14 +351,16 @@ How do I build images?
...
 NOTES:
 
-1. edxapp and IDAs use the ``latest`` tag since their configuration changes have been merged to master branch of
-   ``edx/configuration``.
-2. The elasticsearch used in devstack is built using elasticsearch-devstack/Dockerfile and the ``devstack`` tag.
+1. edxapp and IDAs use the ``latest`` tag for configuration changes which have been merged to master branch of
+   their repository and ``edx/configuration``.
+2. Images for a named Open edX release are built from the corresponding branch
+   of each repository and tagged appropriately, for example ``hawthorn.master``
+   or ``hawthorn.rc1``.
```
Comment (nedbat) [inline README.rst:363]: For devstack, people should probably always be on a ".master" branch.  But for their own development, they will need to be on their own fork, right?  Do we have a way to support that?

File: README.rst [hunk 2]
```diff
@@ -385,6 +387,21 @@ For example, if you wanted to build tag ``release-2017-03-03`` for the
 E-Commerce Service, you would modify ``ECOMMERCE_VERSION`` in
 ``docker/build/ecommerce/ansible_overrides.yml``.
 
+How do I run the images for a named Open edX release?
+-----------------------------------------------------
+
+1. Set the OPENEDX_RELEASE environment variable to the appropriate image tag;
+   "hawthorn.master", "zebrawood.rc1", etc.
```
Comment (nedbat) [inline README.rst]: "Note that unlike a server install, OPENEDX_RELEASE should not have the 'open-release/' prefix."

File: README.rst [hunk 3]
```diff
@@ -385,6 +387,21 @@ For example, if you wanted to build tag ``release-2017-03-03`` for the
...
+2. Check out the correct branch (such as ``open-release/hawthorn.master``) in
+   the local checkout of each service repository.  ``make dev.checkout``
+   should handle this for you once you've set the environment variable above.
```
Comment (nedbat) [inline README.rst]: Can we say this as "Use `make dev.checkout` to check out the correct branch in the local checkout of each service repository."  ?  Is there ever a reason they'd want to not use dev.checkout?

Comment (nedbat) [reply README.rst]: Also, "once you've set the OPENEDX_RELEASE environment variable above."

File: README.rst [hunk 4]
```diff
@@ -385,6 +387,21 @@ For example, if you wanted to build tag ``release-2017-03-03`` for the
...
+3. ``make pull`` to get the correct images.
+
+All ``make`` targes and ``docker-compose`` calls should now use the correct
```
Comment (nedbat) [inline README.rst]: "targes"

File: docker-compose.yml [hunk 5]
```diff
@@ -24,7 +24,7 @@ services:
 
   elasticsearch:
     container_name: edx.devstack.elasticsearch
-    image: edxops/elasticsearch:devstack
+    image: edxops/elasticsearch:${OPENEDX_RELEASE:-latest}
```
Comment (nedbat) [inline docker-compose.yml]: This had been "devstack", but now defaults to "latest"

**Review by nedbat — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (jmbowman) [reply README.rst:363]: Most changes can still use the pre-built images for the branch they were derived from.  If they make more extensive changed to dependencies or such, they can build their own images using the instructions above.

Comment (nedbat) [reply README.rst:363]: Oh, right.  I keep getting confused between pulling images, and pulling code.

**Review by nedbat — APPROVED** [review]

**Pull request merged**